src/bin/club-create

```diff
@@ -30,70 +22,34 @@ program
     .option('-s, --state [id|name]', 'Set workflow state of story', '')
     .option('-y, --type [name]', 'Set type of story, default: feature', 'feature')
     .parse(process.argv);
-spin.setSpinnerString(27);
 
 const main = async () => {
+    const entities = await storyLib.fetchEntities();
     if (!program.idonly)
         spin.start();
-    try {
-        [ wfs, members, projects, epics ] = await Promise.all([
-            client.listWorkflows(),
-            client.listMembers(),
-            client.listProjects(),
-            client.listEpics(),
-        ]);
-        wf = wfs[0];    // TODO: this is always getting the default workflow
-    } catch (e) {
-        log('Error fetching workflows');
-    }
     let update = {
         name:          program.title,
         story_type:    program.type,
         description:   program.description,
         estimate:      program.estimate || undefined
     };
     if (program.project) {
-        const projectMatch = new RegExp(program.project, 'i');
-        const project = projects.filter(s => {
-            return !!`${s.id} ${s.name}`.match(projectMatch);
-        })[0];
-        update.project_id = (project || {}).id;
+        update.project_id = (storyLib.findProject(entities, program.project) || {}).id;
     }
     if (program.state) {
-        const stateMatch = new RegExp(program.state, 'i');
-        const newState = wf.states.filter(s => {
-            return !!`${s.id} ${s.name}`.match(stateMatch);
-        })[0];
-        update.workflow_state_id = (newState || {}).id;
+        update.workflow_state_id = (storyLib.findState(entities) || {}).id;
     }
     if (program.epic) {
-        const epicMatch = new RegExp(program.epic, 'i');
-        const epic = epics.filter(s => {
-            return !!`${s.id} ${s.name}`.match(epicMatch);
-        })[0];
-        update.epic_id = (epic || {}).id;
+        update.epic_id = (storyLib.findEpic(entities, program.epic) || {}).id;
     }
     if (program.estimate) {
         update.estimate = parseInt(program.estimate, 10);
     }
     if (program.owners) {
-        const ownerMatch = new RegExp(program.owners.split(',').join('|'), 'i');
-        update.owner_ids = members.filter(m => {
-            return !!`${m.id} ${m.profile.name} ${m.profile.mention_name}`
-                .match(ownerMatch);
-        }).map(m => m.id);
+        update.owner_ids = storyLib.findOwners(entities, program.owners);
     }
     if (program.label) {
-        const labelMatch = new RegExp(program.label.split(',').join('|'), 'i');
-        try {
-            const labels = await client.listResource('labels');
-            update.labels = labels.filter(m => {
-                return !!`${m.id} ${m.name}`
-                    .match(labelMatch);
-            }).map(m => { return { name: m.name }; });
-        } catch (e) {
-            log('Failed to fetch labels to attach.');
-        }
+        update.labels = storyLib.findLabels(program.label);
```

Comment:
I think this requires us to `await` the response...

Reply:
Since this is now in `fetchEntities` async is not necessary.